### DISCUSSION TITLE
How about moving this project to the github.com/rhino organization

### DISCUSSION THREAD

**Discussion by p-bakker** [mozilla/rhino#1230] [Ideas]
Hi all,

I've managed to secure the https://github.com/rhino organization and I think it might be benefitial to move the project there, for a number of reasons:
- moves us away from under the Mozilla organization, which is confusing (to some) given that Mozilla isn't actively involved in the project anymore, moved it to the graveyard and tossed out all the Rhino documentation (without notification)
- allows us to put stuff in dedicated repo's, like:
  - moving the docs into their own repo and hosting them on https://rhino.github.io
  - moving the Kangax compat and supporting code into its own repo
  - maybe splitting out the tool (in toolsrc) off into their own repo
  - collaborate on other things (like a [DAP](https://microsoft.github.io/debug-adapter-protocol/) Server or standardize web features for serverside JavaScript runtimes (see https://github.com/mozilla/rhino/discussions/1224)
- allow more fine-grained control over who has commit access (if different parts of the code are split over different repo's)
- ...

For now I just forked the repo to there, but that one should be deleted and then this repo should be [transferred](https://docs.github.com/en/enterprise-server@3.3/repositories/creating-and-managing-repositories/transferring-a-repository) if we want to make proper use of the rhino org.

I've secured the rhino org on GitHub with the agreement that if we end up not using it, it'll be returned to the previous owner

So, let me know your thoughts. I think it would be a nice step forward for the project

FYI: while I'm the current owner of the rhino org, I have no problem handing that over to the current maintainers of Rhino if that is desired and removing myself as owner

**Comment by rbri** [comment]
Sounds reasonable to me. +1

**Comment by botic** [comment]
I had some troubles looking up details in the documentation over the holidays. So I'm asking if there is still some progress going on at this problem? Since new coworkers also had this experience, I think this is a real pain point for new users of Rhino / projects building on top of Rhino.

**Reply by p-bakker** [reply]
As per https://github.com/mozilla/rhino/issues/954#issuecomment-949763810, you can find a version of the WIP of moving the docs at https://p-bakker.github.io/rhino/

As for moving to the github.com/rhino org: only got a few responses to the idea, but so far all positive. However, only 1 of the maintainers of Rhino responded.

Note that getting some version of the docs up and running again isn't dependant on moving (development over) to the Rhino org

**Reply by p-bakker** [reply]
Or check out https://rhino.github.io

**Comment by p-bakker** [comment]
One (possible?) obstacle for moving to our own organization might be the cost incurred for GitHub Actions and file storage (GitHub Packages mainly I think): under the free GitHub offering there are limits to how much usage is allowed for these things.

With our GitHub CI Action taking +/- 1 hour on each commit, I think we'd start incurring costs at approx. 33 commits a month.

ATM, being hosted under the Mozilla umbrella, I assume Mozilla picks up this tab and not sure if they would still do so if Rhino moved to its own organization.

So we'd need someone to pick up this tab...

**Reply by gmt2001** [reply]
Would not apply unless you are using private repositories

> GitHub Actions usage is free for standard GitHub-hosted runners in public repositories, and for self-hosted runners.
>
> For private repositories, each GitHub account receives a certain amount of free minutes and storage for use with GitHub-hosted runners, depending on the product used with the account. Any usage beyond the included amounts is controlled by spending limits.

https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions

Same for GitHub Packages

https://docs.github.com/en/billing/managing-billing-for-github-packages/about-billing-for-github-packages

**Reply by p-bakker** [reply]
Reread the docs and I think you're right, which is great!

**Comment by gbrail** [comment]
I like this idea in general, and I'm glad that you secured the org!

I'm also thinking more seriously that I can find some time to work on Rhino if we're willing to make some more dramatic changes that would affect backward compatibility. 

Would changing the repo and presumably the Maven package name be a good opportunity for that?

**Reply by p-bakker** [reply]
Personally I think I rather keep the source for a backwards compatible version of Rhino and a new , non-backwards compatible v2 version in the same repo, but a different Maven package name for a new non-backwards compatible version is a good idea I think

**Reply by p-bakker** [reply]
Thoughts @gbrail? 

If I look at https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository, transferring the rhino repo from the github.com/mozilla to the github.com/rhino organization seems like a pretty straightforward ordeal and when we do everything, including any old links remain working like before

I from my pov we could transfer the repo today

**Reply by p-bakker** [reply]
And wrt moving the maven packages: think that can also be done today

The rhino artifacts used to live under https://mvnrepository.com/artifact/rhino, dunno if we could resurrect that group (no idea who controls that) or what other group would make sense

**Comment by gbrail** [comment]
Progress on this: 

The Mozilla people are happy to support this, but as a prerequisite we have to figure out who legally owns the Rhino IP. See https://github.com/mozilla/rhino/discussions/1424 for a discussion on this.
